src/model/consent/consent.ts

```diff
@@ -83,20 +83,44 @@ export class ConsentDB {
   }
 
   // Update Consent
-  // No validation against Null or illegal updates in models
+  // Only validating against updating existing fields
   public async update (consent: Consent): Promise<number> {
     // Returns number of updated rows
-    const updateCount: number = await this
-      .Db<Consent>('Consent')
-      .where({ id: consent.id })
-      .update(consent)
-
-    // Ensure that the caller knows that the resource does not exist
-    if (updateCount === 0) {
-      throw new NotFoundError('Consent', consent.id)
-    }
-
-    return updateCount
+    // Transaction to make the update atomic
+    return this.Db.transaction(async (trx): Promise<number> => {
+      // Transaction is rolled back automatically if there is
+      // an error and the returned promise is rejected
+      const consents: Consent[] = await trx<Consent>('Consent')
+        .select('*')
+        .where({ id: consent.id })
+        .limit(1)
+
+      if (consents.length === 0) {
+        throw new NotFoundError('Consent', consent.id)
+      }
+
+      const existingConsent = consents[0]
+      const updatedConsent: Consent = { ...consent }
+
+      // Cannot update fields once they have been added properly to the DB
```

Comment:
what do you mean added properly?

Reply:
When fields are at their final or "proper" values. This mainly concerns `credentialStatus` which has a slightly different constraint.

Reply:
Done